core/arch/arm/pta/sdp_pa.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2017, ARM Limited, All Rights Reserved
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <kernel/pseudo_ta.h>
+#include <mm/core_memprot.h>
+#include <string.h>
+
+#include <sdp_pa.h>
+
+#define PTA_NAME "sdp_pa.pta"
+
+/*
+ * Specficed client TA UUID which has the access to this pta
+ */
+#define TA_SPECFICED_UUID { 0x0b7a14e0, 0xb667, 0x4b3d, \
+		{ 0x84, 0x04, 0xc3, 0xd0, 0xf8, 0xdc, 0x12, 0x44 } }
+
+/*
+ * Specificed TA needs to create pagetable item for data
+ * put in SDP memory. Create a PTA for specficed TA to retrieve
+ * SDP memory physical address.
+ */
+static TEE_Result sdp_pa_cmd_virt_to_phys(uint32_t types,
+			     TEE_Param params[TEE_NUM_PARAMS])
+{
+	char *src = params[0].memref.buffer;
+	size_t sz = params[0].memref.size;
+	paddr_t phys;
+
+	if (types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+				     TEE_PARAM_TYPE_VALUE_OUTPUT,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE)) {
+		EMSG("bad parameters types: %x", (unsigned)types);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	if (!core_vbuf_is(CORE_MEM_SDP_MEM, src, sz)) {
+		EMSG("bad memref secure attribute");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	phys = virt_to_phys(src);
+
+	params[1].value.a = (uint32_t)phys;
+	params[1].value.b = (uint32_t)(phys >> 32);
+
+	return TEE_SUCCESS;
+}
+
+/*
+ * Trusted Application Entry Points
+ */
+
+static TEE_Result create_ta(void)
+{
+	DMSG("create entry point for pseudo TA \"%s\"", PTA_NAME);
+	return TEE_SUCCESS;
+}
```

Comment:
minor: if these functions are empty, you can simply not define them.

Reply:
OK.